src/components/Weather.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react';
+import Moment from 'react-moment';
+import axios from 'axios';
+import AlertGeneral from '../containers/AlertGeneral/AlertGeneral';
+import FetchWeather from '../services/FetchWeather';
+
+//for time: time.slice(14,19)
+
+const weatherUrl = 'http://localhost:4001'
+
+export class Weather extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      nameCity: '',
+      forecast: [],
+      alertCity: '',
+      listCity: '',
+      alertTitle: '',
+      alertDescription: ''
+    }
+  }
+
+  componentDidMount() {
+    this.getWeather();
+    this.getCity();
+  } 
+  
+  getWeather = async () => {
+    const city = this.state.nameCity === '' || this.state.nameCity === undefined ?
```

Comment:
This code is correct but very long, since both you conditions are falsy values, you can rewrite the code like this:
`const city = this.state.name || 'Barcelona`